cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,74 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import SettingsPageObject from '../support/pages/settings.pageObject';
+import { faker } from '@faker-js/faker';
+
+const settingsPage = new SettingsPageObject();
+let user;
+
 describe('Settings page', () => {
   before(() => {
+    cy.task('db:clear');
+    user = {
+      email: faker.internet.email(),
+      username: faker.internet.userName(),
+      password: faker.internet.password(10),
+      bio: faker.lorem.sentence()
+    };
 
+    cy.register(user.email, user.username, user.password);
   });
 
   beforeEach(() => {
-
+    cy.login(user.email, user.username, user.password);
+    settingsPage.visit();
   });
 
-  it('should provide an ability to update username', () => {
+  it('should provide an ability to update bio', () => {
+    const newBio = faker.lorem.sentences(2);
+    settingsPage.typeBio(newBio);
+    settingsPage.submitForm();
 
+    settingsPage.assertProfilePage();
+    cy.get('[data-cy=bio-settings]').should('contain.text', newBio);
   });
 
-  it('should provide an ability to update bio', () => {
+  it('should provide an ability to update username', () => {
+    const newUsername = faker.internet.userName();
+    settingsPage.typeUsername(newUsername);
+    settingsPage.submitForm();
 
+    settingsPage.assertProfilePage(newUsername);
+    cy.get('[data-cy=username-settings]').should('contain.text', newUsername);
+    user.username = newUsername;
   });
 
-  it('should provide an ability to update an email', () => {
+  it('should provide an ability to update email', () => {
+    const newEmail = faker.internet.email();
+    settingsPage.typeEmail(newEmail);
+    settingsPage.submitForm();
+
+    settingsPage.assertProfilePage(user.username);
```

Comment:
Ensure that `assertProfilePage` correctly verifies that the profile page is displayed with the correct username after updating the email. This will confirm that the email update was successful.